sanic/response/convenience.py

```diff
@@ -226,16 +230,25 @@ async def file(
 
     filename = filename or path.split(location)[-1]
 
-    async with await open_async(location, mode="rb") as f:
-        if _range:
-            await f.seek(_range.start)
-            out_stream = await f.read(_range.size)
-            headers[
-                "Content-Range"
-            ] = f"bytes {_range.start}-{_range.end}/{_range.total}"
-            status = 206
-        else:
-            out_stream = await f.read()
+    try:
```

Comment:
This function is extremely long. Might be a good time to refactor into smaller functions before growing it even more.

Reply:
Nice idea. I'll make a separate PR for introducing `FileResponse` similar to our recent `JSONResponse`.